### ISSUE TITLE
Flux hangs when parsing array with missing commas in v2.0.3

### ISSUE THREAD

**Issue by bemasher** [influxdata/influxdb#20335]
__Steps to reproduce:__

1. Open a task for editing.
2. Make edits.

or:

1. Execute query from cli `influx query -f query.flux`

__Expected behavior:__
 * Edits happen, everything is happy.
 * Query executes.

__Actual behavior:__
The Chrome tab in which the editor is running hangs. I am unable to record any performance traces or heap snapshots because the tab hangs.

When executing a query Influxd pins a single cpu core and continues allocating memory unbounded until systemd kills.

Anecdotally I'm noticing similar behavior in the flux plugin for vscode (and am having doubts that this issue really belongs on the influxdb repo.)

__Environment info:__
* System info: Windows 10, 19041.685 (Chrome issues are on this machine)
* System info: Linux 5.9.14-arch1-1 x86_64
* InfluxDB version: InfluxDB 2.0.3 (git: fe04d346df) build_date: 2020-12-15T01:00:16Z
* Other relevant environment details:
  * Chrome 87.0.4280.88 (Official Build) (64-bit)
  * vscode 1.52.0 940b5f4bb5fa47866a54529ed759d95d09ee80be x64
  * vsflux v0.5.13

__Performance:__
The following profiles were collected while executing the query listed at the end of this issue. Execution never completes, influx never gets to a point where I would expect it to complain that `task.every` isn't defined. Doesn't appear to matter if the query is valid either.

[profiles.tar.gz](https://github.com/influxdata/influxdb/files/5693309/profiles.tar.gz)
[vars.txt](https://github.com/influxdata/influxdb/files/5693310/vars.txt)
[iostat.txt](https://github.com/influxdata/influxdb/files/5693307/iostat.txt)

```flux
srcData = (fn, lbl) => from(bucket: "7d")
    |> range(start: -task.every))
    |> filter(fn: (r) => (
        r._measurement == "environment" and (
            r._field == "temperature" or
            r._field == "humidity" or
            r._field == "pressure" or
            r._field == "vbat"
        )
    ))
    |> aggregateWindow(every: 1h, fn: fn)
    |> map(fn: (r) => ({ r with
        _field: "${r._field}_${lbl}"
    }))

union(tables: [
    srcData(fn: max, lbl: "max")
    srcData(fn: min, lbl: "min")
    srcData(fn: median, lbl: "median")
])
// |> to(bucket: "90d")
```

**Comment by danxmoran** [comment]
Hi @bemasher, thanks for the report. The `profiles.tar.gz` file you uploaded just contains the string:
```
Unknown profile
```
If you're able to easily reproduce, could you re-profile and re-upload the result? I'll dig in from my end at the same time.

**Comment by bemasher** [comment]
@danxmoran, looks like the url provided in the issue template doesn't point to a valid pprof endpoint. I've attached a tar of profiles for heap, allocs, and profile while the query was executing.

[profiles.tar.gz](https://github.com/influxdata/influxdb/files/5699579/profiles.tar.gz)

**Comment by bemasher** [comment]
I generated a cpu profile from running the same query through the flux compile command. The majority of the cpu time is spent in a cgo calls from libflux's flux_analyze, and flux_parse. See attached profile and pdf.

[profile002.pdf](https://github.com/influxdata/influxdb/files/5700854/profile002.pdf)
[cpu.pprof.gz](https://github.com/influxdata/influxdb/files/5700856/cpu.pprof.gz)

**Comment by danxmoran** [comment]
Thanks @bemasher. Were you able to parse this query successfully on a previous 2.x version? Trying to bisect the flux change that might be causing the behavior.

**Comment by bemasher** [comment]
This query does compile (or at least fail gracefully) in v0.98.0 of flux. If the query is corrected by removing the extra `)` at the end of the range statement, and adding commas in the tables argument of union, it also compiles in v0.99.0.

Correcting only the extra parenthesis but not the commas does still produce the same issue in v0.99.0. The missing commas appear to be what flux is choking on. The following version of the query does compile in v0.98.0 and v0.99.0:

```
srcData = (fn, lbl) => from(bucket: "7d")
    |> range(start: -task.every)
    |> filter(fn: (r) => (
        r._measurement == "environment" and (
            r._field == "temperature" or
            r._field == "humidity" or
            r._field == "pressure" or
            r._field == "vbat"
        )
    ))
    |> aggregateWindow(every: 1h, fn: fn)
    |> map(fn: (r) => ({ r with
        _field: "${r._field}_${lbl}"
    }))

union(tables: [
    srcData(fn: max, lbl: "max"),
    srcData(fn: min, lbl: "min"),
    srcData(fn: median, lbl: "median")
])
```

**Comment by danxmoran** [comment]
I was wondering why I couldn't reproduce it; my editor auto-inserted the commas :facepalm:

I'm able to reproduce via `influx query`, thanks for all the debugging info. Will follow up with the Flux team to see what's going on here.

**Comment by danxmoran** [comment]
@bemasher this has been fixed in Flux, the patch will be included in version 2.0.4

**Comment by danxmoran** [comment]
Closed by #20678
